src/cache.js

```diff
@@ -0,0 +1,67 @@
+const Apigee = require('./lib/apigee')
+const yaml = require('js-yaml')
+const fs = require('fs')
+const chai = require('chai')
+const expect = chai.expect
+
+const isUpdated = (a, b, properties) => {
+  return properties.find(prop => {
+    if (prop.includes('.')) {
+      const props = prop.split('.')
+      return a[props[0]][props[1]] !== b[props[0]][props[1]]
+    }
+    return a[prop] !== b[prop]
+  })
+}
+
+const validateDate = (cache) => {
+  if ((cache.timeoutInSec !== null && cache.timeOfDay !== null) ||
+     (cache.timeOfDay !== null && cache.expiryDate !== null) ||
+     (cache.timeoutInSec !== null && cache.expiryDate !== null)) {
+      console.log('You can only have 1 expirySettings option. Set to default 300 seconds')
+    return {'timeoutInSec': {'value': 300}, 'valuesNull': cache.expirySettings.valuesNull}
```

Comment:
cache.expirySettings might be undefined, should check if it's defined before using .valuesNull